util/src/main/scala/scala/scalanative/io/package.scala

```diff
@@ -14,4 +16,13 @@ package object io {
     try f(buffer)
     finally pool.reclaim(buffer)
   }
+
+  def packageNameFromPath(path: Path, extension: String): String = {
```

Comment:
It seems to me that the `extension` parameter is redundant. What do you think about this?

```scala
  def packageNameFromPath(path: Path): String = {
    val fileName = path.getFileName.toString
    val base     = fileName.split('.').init.mkString(".")

    Option(path.getParent) match {
      case Some(parent) => parent.resolve(base).asScala.mkString(".")
      case None         => base
    }
  }
```